### PR TITLE
Part a

### PR DESCRIPTION
**Pull request by sdam12**
Please review and provide your feedback.Thank You

### CODE REVIEW

**Review by abhinavachanta — CHANGES_REQUESTED** [review]
Please check comments which i added to .cpp files in test folder. Some header file locations could not be resolved.

**Comment by sdam12** [comment]
The Changes are taken into consideration. And also updated in the github branch. Please check and let us know if anything else required. Thank You

**Comment by jrajeev459** [comment]
The changes made in the make file were good.

File: makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	$(CC) -g -c $(CFLAGS) $(INCLUDECADMIUM) test/src/receiver/main.cpp -o build/receiver_main.o
+subnet_main.o: test/src/subnet/main.cpp
+	$(CC) -g -c $(CFLAGS) $(INCLUDECADMIUM) test/src/subnet/main.cpp -o build/subnet_main.o	
+
+../message.o: 
+	$(CC) -g -c $(CFLAGS) $(INCLUDECADMIUM) src/message.cpp -o build/message.o
+	
+clean:
+	rm -f bin/*.exe build/*.o *~
+	-for d in src; do (cd $$d; rm -f *.o *~ ); done
```
Comment (cruizm) [inline makefile]: This line is not needed

**Review by cruizm — CHANGES_REQUESTED** [review]

**Review by cruizm — APPROVED** [review]

**Comment by abhinavachanta** [comment]
Please update the README file according to the changes in folder structure

**Comment by sdam12** [comment]
> Please update the README file according to the changes in folder structure

Changes in the Readme.txt file is done. Thank you for feedback.

File: makefile (hunk 1 above)
Comment (sdam12) [reply makefile]: The line is removed from the makefile. Thank You for the feedback.

**Comment by abhinavachanta** [comment]
All the requirements look addressed and the codes were executing successfully

**Review by abhinavachanta — APPROVED** [review]
All the requirements look addressed and the codes were executing successfully.

**Review by jrajeev459 — APPROVED** [review]
Everything is working and the .exe were being created and deleted using the commands from make file.

**Pull request merged**